### ISSUE TITLE
Comments incorrectly appended to hovered functions

### ISSUE THREAD

**Issue by ArchLand64** [JohnnyMorganz/luau-lsp#310]
A common pattern that I do in my code is use large comments to organize all the different types of functions that I write. My problem is that those large comments are incorrectly appended to hovered function documentation, as boxed in red below.
![image](https://user-images.githubusercontent.com/68730440/222977129-9ba6fb81-124b-4a98-9acf-486629314c2f.png)

I expect only the comment directly above the function to be highlighted. In the case of the above picture, I only want to see "Does physics simulation".

Here is my example code:
```lua
--[[
	Helps do math for physics
]]
local function Add(a: number, b: number): number
	return a + b
end

--[[
==========================Physics Functions==========================
]]

--[[
	Does physics simulation
]]
local function _Simulate(a: number, b: number): number
	return Add(a, b) * Add(a, b)
end
```

**Comment by LastTalon** [comment]
To be clear, this is an issue when using doc comments as well. Often floating doc comments will exist such as the following example and it happens there, too. When it happens in this case it's actually worse because it actually displays confusing documentation.

![image](https://github.com/JohnnyMorganz/luau-lsp/assets/2075484/4db6be2d-f8ce-48b1-9970-9f6f4763932f)

**Comment by mxruben** [comment]
I think this might be an issue specifically with multiline comments. I've noticed that the contents of multiline comments get appended to the doc comment below them. If there isn't a doc comment it just becomes one. It seems like multiline comments are just always treated as doc comments?
![Screenshot from 2024-06-21 12-24-08](https://github.com/JohnnyMorganz/luau-lsp/assets/89617289/1473287c-2004-4c5d-a25d-cf4ffa9807dc)

**Comment by m4fh** [comment]
Wouldn't this issue be fixed if we simply ignored multiple documentation comments on a single item? As far as I know, that's how it works in TypeScript, and Moonwave is very JS/TSdoc-like.

I also think this approach would be much easier to implement.

```ts
/**
 * Not a comment for foo.
 */
/**
 * A comment for foo.
 */
type Foo = any
```

![Image](https://github.com/user-attachments/assets/6e7f4c5b-c11a-4972-bcc9-5472c1514e36)

I'm pretty sure this pattern is uncommon, to say the least:

```luau
--[=[
    Not a comment for foo.
]=]
--[=[
    A comment for foo.
]=]
type Foo = any
```

So it shouldn't _really_ break anything.

![Image](https://github.com/user-attachments/assets/fa76572d-1b0e-4ef7-9d05-b625802bac1b)

**Comment by JohnnyMorganz** [comment]
Yeah I think so. The problem is just that we should attach multiple documentation comments if they are single line comments.

The access to comment locations that Luau gives us makes this slightly trickier (but not impossible), just never got round to it
